File: SuperNaviBeaconAPI/SuperNaviBeaconAPI/Models/DtoBeacon.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace SuperNaviBeaconAPI.Models
{
    public class DtoBeacon
    {
        public String uuid { get; set; }
        public int majorid { get; set; }
        public int minorid { get; set; }
        public int rssi { get; set; }
        public String supermarket { get; set; }
        public int positionX { get; set; }
        public int positionY { get; set; }
        public int count { get; set; }

        public Beacon toDomainObject()
        {
            Beacon beacon = new Beacon(this.supermarket, this.uuid, this.majorid, this.minorid, this.positionX, this.positionY)
            {
                rssi = this.rssi,
                count = this.count
            };

            return beacon;
        }
    }

    public class DtoBeaconList
    {
        public List<DtoBeacon> beacons { get; set; }
    }
}

File: SuperNaviApp/app/src/main/java/com/midas/supernavi/Models/DtoBeacon.java
package com.midas.supernavi.Models;

import org.altbeacon.beacon.Beacon;

/**
 * Created by hshe on 15/10/2016.
 */

public class DtoBeacon {

    public DtoBeacon(Beacon beacon){
        this.uuid = beacon.getId1().toString();
        this.majorid = beacon.getId2().toInt();
        this.minorid = beacon.getId3().toInt();
        this.rssi = beacon.getRssi();
        this.positionX = 0;
        this.positionY = 0;
    }

    public String uuid;
    public int majorid;
    public int minorid;
    public int rssi;
    public String supermarket;
    public int positionX;
    public int positionY;
    public int count = 0;
}
